src/js/core/player.js

```diff
@@ -792,17 +792,53 @@ function name(): {country: string, firstName: string, lastName: string} {
     const fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
     const firstNameRow = playerNames.first[country].find(row => row[1] >= fnRand);
     if (firstNameRow === undefined) {
-        throw new Error(`Undefined firstNameRow (fnRand=${fnRand}`);
+        throw new Error(`Undefined firstNameRow (fnRand=${fnRand})`);
+    }
+    let firstName = firstNameRow[0];
+
+    // Middle Name, randomly
+    const middle = random.uniform(0, 1);
+    if (middle < 0.05) {
+        const mnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+        const middleNameRow = playerNames.first[country].find(row => row[1] >= mnRand);
+        if (middleNameRow === undefined) {
+            throw new Error(`Undefined middleNameRow (mnRand=${mnRand})`);
+        }
+        let middleName = middleNameRow[0];
+
+        // abbreviation rules:
+        // 50% will initialize their middle name
+        //   e.g. {Xavier Y. Zed}
+        // 50% will use it fully
+        //   e.g. {Xavier Yanick Zed}
+        const abbrev = random.uniform(0, 1);
+        if (abbrev < 0.5 || country === "USA") {
+            // initialize it
+            middleName = `${middleName.charAt(0)}.`;
+        } else {
+            // use it fully
+        }
+        firstName = `${firstName} ${middleName}`;
     }
-    const firstName = firstNameRow[0];
 
     // Last name
     const lnRand = random.uniform(0, playerNames.last[country][playerNames.last[country].length - 1][1]);
     const lastNameRow = playerNames.last[country].find(row => row[1] >= lnRand);
     if (lastNameRow === undefined) {
-        throw new Error(`Undefined lastNameRow (lnRand=${lnRand}`);
+        throw new Error(`Undefined lastNameRow (lnRand=${lnRand})`);
+    }
+    let lastName = lastNameRow[0];
+
+    // Jr. or III, randomly
+    const son = random.uniform(0, 1);
+    const english = ["USA", "Canada", "Australia", "New Zealand", "England", "Ireland", "Jamaica", "Phillippines"];
+    if (english.indexOf(country) > 0 && son < 0.01) {
+        if (son < 0.004) {
+            lastName = `${lastName} III`;
```

Comment:
Serious question:

do you think this shouldn't be random, but rather I should look at if anyone in the league history has been born with the same first-last combo within the last 50 years and add +1 to Jr, III, IV, etc for every consecutive John Smith within ~1 generation?

E.g. we randomize a John Smith in 2024 and another John Smith appears >10 years later but less than 30 years later, so we make the second John Smith: John Smith Jr. If another one shows up then it's John Smith III but if none show up within 30 years then the "counter" goes back to just normal John Smith.

I'm struggling to estimate whether that would have too many Jr's or too few, haha.

If we started grabbing "hometowns" in randomization then we could limit this to Jr's within the same city (or same city as the 'fathers' team maybe, how do families work?) but that is waaaaaaaay out of scope for just adding middle names.